mujoco_warp/_src/collision_driver.py

```diff
@@ -72,30 +227,42 @@ def _sphere_filter(
   geom2: int,
   worldid: int,
 ) -> bool:
+  # 1: plane
+  # 2: sphere
+  # 4: aabb
+  # 8: obb
+
+  center1 = geom_aabb[geom1, 0]
+  center2 = geom_aabb[geom2, 0]
+  size1 = geom_aabb[geom1, 1]
+  size2 = geom_aabb[geom2, 1]
+  rbound1 = geom_rbound[worldid, geom1]
+  rbound2 = geom_rbound[worldid, geom2]
   margin1 = geom_margin[worldid, geom1]
   margin2 = geom_margin[worldid, geom2]
-  pos1 = geom_xpos_in[worldid, geom1]
-  pos2 = geom_xpos_in[worldid, geom2]
-  size1 = geom_rbound[worldid, geom1]
-  size2 = geom_rbound[worldid, geom2]
+  xpos1 = geom_xpos_in[worldid, geom1]
+  xpos2 = geom_xpos_in[worldid, geom2]
+  xmat1 = geom_xmat_in[worldid, geom1]
+  xmat2 = geom_xmat_in[worldid, geom2]
+
+  if rbound1 == 0.0 or rbound2 == 0.0:
```

Comment:
you're already checking these conditions inside `_plane_filter` and `_sphere_filter`, no need to check them twice?

this could probably just be the four conditions:

```py

if opt_broadphase_fitler & int(BroadphaseFilter.PLANE.value):
  ...
if opt_broadphase_fitler & int(BroadphaseFilter.SPHERE.value):
  ...
if opt_broadphase_fitler & int(BroadphaseFilter.AABB.value):
  ...
if opt_broadphase_fitler & int(BroadphaseFilter.OBB.value):
  ...
```

Reply:
checking for plane geoms is removed from the individual filters. `_broadphase_filter` checks if either geom is a plane and there are two code paths: if one of the geoms is a plane: 1) plane filter, else: 2) sphere filter, aabb filter, obb filter